collector/src/main/java/io/prometheus/jmx/JmxScraper.java

```diff
@@ -135,31 +137,38 @@ private void scrapeBean(MBeanServerConnection beanConn, ObjectName mbeanName) {
         }
         MBeanAttributeInfo[] attrInfos = info.getAttributes();
 
+        Map<String, MBeanAttributeInfo> name2AttrInfo = new LinkedHashMap<String, MBeanAttributeInfo>();
         for (int idx = 0; idx < attrInfos.length; ++idx) {
             MBeanAttributeInfo attr = attrInfos[idx];
             if (!attr.isReadable()) {
                 logScrape(mbeanName, attr, "not readable");
                 continue;
             }
-
-            Object value;
-            try {
-                value = beanConn.getAttribute(mbeanName, attr.getName());
-            } catch(Exception e) {
-                logScrape(mbeanName, attr, "Fail: " + e);
-                continue;
+            name2AttrInfo.put(attr.getName(), attr);
+        }
+        AttributeList attributes;
+        try {
+            attributes = beanConn.getAttributes(mbeanName, name2AttrInfo.keySet().toArray(new String[0]));
+        } catch (Exception e) {
+            throw new RuntimeException("Failed to read attributes " + name2AttrInfo.keySet() + " from " + mbeanName);
```

Comment:
Handle things like this gracefully, many JMX servers are broken and attributes can change